databaseManager.class.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+    include_once 'databaseManagerInterface.php';
+
+    /**
+     * Define an Abstract class for all databases
+     * connections to extend
+     */
+    abstract class DatabaseManager implements DatabaseManagerInterface
+    {
+        protected $dbDriver;
+        protected $host;
+        protected $port;
+        protected $username;
+        protected $password;
+        protected $dbName;
+        protected $unixSocket;
+        protected $charset;
+        protected $connection;
+
+        /**
+         * Connect database
+         */
+        abstract protected function openConnection();
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        abstract protected function selectData($table, $conditions = array());
+
+        /**
+         * Insert data into the database
+         * @param string name of the table
+         * @param array the data for inserting into the table
+         */
+        abstract protected function insertData($table,$data);
+        
+        /**
+         * Update data into the database
+         * @param string name of the table
+         * @param array the data for updating into the table
+         * @param array where condition on updating data
+         */
+        abstract protected function updateData($table, $data, $conditions);
+
+        /**
+         * Delete data from the database
+         * @param string name of the table
+         * @param array where condition on deleting data
+         */
+        abstract protected function deleteData($table, $conditions);
```

Comment:
@lumakos Why these abstract methods aren't described by the interface?

Reply:
Actually, I have to described all these abstract methods to the interface, it's my fault.